web/src/click_handlers.js

```diff
@@ -927,6 +927,25 @@ export function initialize() {
         settings_panel_menu.mobile_deactivate_section();
     });
 
+    // When tabbing through an overlay, and the tab gets to the last element to receive focus
+    // this code gets the last element in the overlay and calls an onfocus event on it which in turn
+    // returns the focus to the first element in the overlay (the overlay container) and starts the tab cycle
+    // over again prevent tabbing from going outside of the overlay and focusing on outside elements
+    // In order to make this reusable, a data attribute (data-top-focus-target) was attached to the
+    // element that returns focus to the top element in the overlay. All that is required to reuse this
+    // is to get the span with the class (return-tab-focus-to-modal-top) and drop it in an overlay
+    // then set the value of the "data-top-focus-target" to the id of the overlay container or a top element
+    // that can start the tab cycle
+
+    $(".return-tab-focus-to-modal-top").on("focus", () => {
+        const $top_focus_target_element = $("[data-top-focus-target]").data("top-focus-target");
```

Comment:
Okay. I think this should still be a resusable function, and not just a shift of all the code into this anonymous callback.

The bigger issue, though, is I don't see how this listener or the target element it works on is related to a specific modal, once more than one modal takes these classes and data attributes. Some use of `$(this)` seems necessary for that to happen.

Reply:
Hi @karlstolley, Thank you for the feedback. Aside from moving the logic into a reusable function, I don't quite understand what you mean. Could you please explain in more detail what you mean?

Reply:
I think the idea is that these classes and data attributes could be placed on multiple modals. I don't understand in your logic how it knows which specific modal it needs to push focus to. `$top_focus_target_element` could be a whole bunch of different modals in the DOM; you need to somehow refer to the active one, again, using `$(this)`, most likely.